src/commands/game_options/spotify.ts

```diff
@@ -225,139 +242,174 @@ export default class SpotifyCommand implements BaseCommand {
-            }
+                )} | Invalid URL in updateOption. playlistURL = ${playlistURL}`,
+            );
 
-            logger.info(
-                `${getDebugLogHeader(messageContext)} | Matched ${
-                    matchedPlaylist.metadata.matchedSongsLength
-                }/${matchedPlaylist.metadata.playlistLength} (${(
-                    (100.0 * matchedPlaylist.metadata.matchedSongsLength) /
-                    matchedPlaylist.metadata.playlistLength
-                ).toFixed(2)}%) Spotify songs`,
+            sendErrorMessage(
+                messageContext,
+                {
+                    title: i18n.translate(
+                        messageContext.guildID,
+                        "command.spotify.invalidURL.title",
+                    ),
+                    description: i18n.translate(
+                        messageContext.guildID,
+                        "command.spotify.invalidURL.description",
+                    ),
+                },
+                interaction,
             );
+            return;
+        }
+
+        let playlistID: string;
+        if (isFullURL) {
+            playlistID = playlistURL.split(SPOTIFY_BASE_URL)[1];
+        } else {
+            try {
+                const response = await fetch(playlistURL);
+                const body = await response.text();
+                playlistID = body.split(SPOTIFY_BASE_URL)[1].split("?si=")[0];
+            } catch (err) {
+                logger.error(
+                    `${getDebugLogHeader(
+                        messageContext,
+                    )} | Failed to get playlist ID from shorthand Spotify URL. playlistURL = ${playlistURL}. err = ${err}`,
+                );
 
-            if (matchedPlaylist.matchedSongs.length === 0) {
                 sendErrorMessage(
                     messageContext,
                     {
                         title: i18n.translate(
-                            guildID,
-                            "command.spotify.noMatches.title",
+                            messageContext.guildID,
+                            "command.spotify.invalidURL.title",
                         ),
                         description: i18n.translate(
-                            guildID,
-                            "command.spotify.noMatches.description",
+                            messageContext.guildID,
+                            "command.spotify.invalidURL.description",
                         ),
                     },
                     interaction,
                 );
-
                 return;
             }
+        }
 
-            await guildPreference.setSpotifyPlaylistID(playlistID);
-
-            await LimitCommand.updateOption(
-                messageContext,
-                0,
-                matchedPlaylist.metadata.matchedSongsLength,
-                undefined,
-                false,
-            );
+        if (playlistID.includes("?si=")) {
```

Comment:
Can we not use a regex to extract the playlist id instead?

Reply:
This check will only apply to the normal Spotify url. I'll move it into there